### ISSUE TITLE
autocompletion popup

### ISSUE THREAD

**Issue by aaltland** [SublimeText/LaTeXTools#678]
I must be doing something wrong:

I don't have the cwl package installed, and set
"command_completion": "never"

in my user preferences. Nevertheless, a popup window keeps appearing the moment I press '\'.

Any ideas what's wrong?

**Comment by r-stein** [comment]
The `command_completion` setting sets whether to insert the completions entries into the autocomplete-popup and not when the popup should be shown.
The autocomplete is shown, because `\` is automatically added into the `auto_complete_triggers` in the user settings if the cwl package is installed. You can open the user settings and remove the entry, but it will be added when you start ST again. Hence everything works as expected, but maybe not as you desire.

If I understand you correct you want to use cwl autocompletion, but it should not automatically be triggered?

**Comment by aaltland** [comment]
Thanks for the ultra-swift response! I had misunderstood the logics behind the popup mechanism. Ideally, I would like to get rid of the popup (which I find truly distracting) at all. And I do not have cwl installed. (I had it once, but removed it.)

A.

**Comment by aaltland** [comment]
p.s.: just removed “\” from auto_complete_triggers in my user settings. This seems to have done the trick. It did not re-appear on restart. Perhaps it was a remnant of my previously installed cwl. A.

**Comment by r-stein** [comment]
Ok then you can just open `Preferences >>> Settings - User ` and delete
``` js
		{
			"characters": "\\",
			"selector": "text.tex.latex"
		}
```
under the setting `auto_complete_triggers`

**Comment by aaltland** [comment]
yep, I had done just this! Thanks, A.

**Comment by ig0774** [comment]
We should probably change the logic to add the auto-complete only if the completion level is not "never". Its probably un-intuitive that these two aspects are entirely disconnected... I'll add it to the TODO list.

**Comment by r-stein** [comment]
This would mean to remove the the auto-complete trigger, because it is only added once.

**Comment by ig0774** [comment]
Right, but ST settings have some sort of observer functionality so you can have a function called when a specific setting is changed. We haven't used it so far because we haven't really had a need to, but this would be an instance where it might make sense.
